### PR TITLE
Use tagified traits input for compendium browser feat search

### PR DESCRIPTION
**Pull request by stwlam**
Bonus: Fix compendium browser feat search launched from PC sheet

### CODE REVIEW

File: src/module/actor/character/feats.ts [hunk 1]
```diff
@@ -42,15 +43,27 @@ class CharacterFeats extends Collection<FeatCategory> {
...
+                ? `hb_${classSlug}`
+                : null;
+
         this.createCategory({
             id: "class",
             label: "PF2E.FeatClassHeader",
-            featFilter: "classes-" + actor.class?.slug,
-            supported: ["class"],
+            featFilter: classTrait ? `traits-${classTrait},traits-archetype` : null,
+            supported: ["class", "archetype"],
```
Comment (CarlosFdez) [inline src/module/actor/character/feats.ts:66]: Interestingly enough, this doesn't do anything. archetype resolves into either skill or class depending on whether it has the skill trait, according to isFeatValid(). You can change that behavior if you want, but we'd end up allowing archetype skill feats in class/archetype feat slots.

File: src/module/actor/character/feats.ts [hunk 2]
```diff
@@ -42,15 +43,27 @@ class CharacterFeats extends Collection<FeatCategory> {
         this.createCategory({
             id: "ancestry",
             label: "PF2E.FeatAncestryHeader",
-            featFilter: "ancestry-" + actor.ancestry?.slug,
+            featFilter: actor.system.details.ancestry?.trait
+                ? `ancestry-${actor.system.details.ancestry?.trait}`
+                : null,
```
Comment (CarlosFdez) [inline src/module/actor/character/feats.ts:48]: Later on it'd be nice to do versatile heritages, but I guess there's no need to rush that yet.

Comment (stwlam) [reply src/module/actor/character/feats.ts:48]: Yeah, I thought about that, but it was a detour while on the detour.

File: src/module/actor/character/feats.ts (hunk 1 above)
Comment (stwlam) [reply src/module/actor/character/feats.ts:66]: archetype skill feats are of the "skill" feat type. Incidentally, we don't really need to have a feat type for anything except features.

File: src/module/actor/character/sheet.ts [hunk 3]
```diff
@@ -746,37 +748,37 @@ class CharacterSheetPF2e extends CreatureSheetPF2e<CharacterPF2e> {
...
         $formulas.find(".daily-crafting").on("click", async () => await this.actor.performDailyCrafting());
 
-        PCSheetTabManager.initialize(this.actor, $html.find<HTMLAnchorElement>('a[data-action="manage-tabs"]')[0]);
+        PCSheetTabManager.initialize(this.actor, $html.find<HTMLAnchorElement>("a[data-action=manage-tabs]")[0]);
 
         // Feat Browser shortcut links
-        $html.find(".feat-browse").on("click", (event) => this.onClickBrowseFeatCompendia(event));
+        for (const link of Array.from(html.querySelectorAll<HTMLAnchorElement>(".feat-browse"))) {
+            link.addEventListener("click", this.onClickBrowseFeatCompendia);
+        }
```
Comment (CarlosFdez) [inline src/module/actor/character/sheet.ts:756]: I'm starting to see why jquery still exists. DOM api is really tedious, and node list in general are kinda painful. Is there any lightweight lib you're aware of to not deal with all this garbage?

Comment (stwlam) [reply src/module/actor/character/sheet.ts:756]: we could acquiesce to the `NodeList#forEach`

File: src/module/actor/character/feats.ts (hunk 1 above)
Comment (CarlosFdez) [reply src/module/actor/character/feats.ts:66]: archetype class feats are of the "class" feat type :)

Comment (stwlam) [reply src/module/actor/character/feats.ts:66]: naw

Comment (stwlam) [reply src/module/actor/character/feats.ts:66]: cf.,
- https://github.com/foundryvtt/pf2e/blob/master/packs/data/feats.db/sentinel-dedication.json
- https://github.com/foundryvtt/pf2e/blob/master/packs/data/feats.db/steel-skin.json

**Pull request merged**